main.tf

```
@@ -124,6 +124,7 @@ resource "google_bigquery_table" "view" {
   dataset_id          = google_bigquery_dataset.main.dataset_id
   friendly_name       = each.key
   description         = each.value["description"]
+  schema              = lookup(each.value, "schema", null)
```

No need to do validation at the module level. This code should be enough.
```suggestion
  schema              = each.value["schema"]
```

---

Changed, and tested here:
https://github.com/cbsi-dto/data-warehouse-ddl/pull/1138#issuecomment-1204466628